StyleCop.Analyzers/StyleCop.Analyzers/NamingRules/SA1311CodeFixProvider.cs

```diff
@@ -43,8 +45,69 @@ public override ImmutableArray<string> GetFixableDiagnosticIds()
                     continue;
 
                 var newName = char.ToUpper(token.ValueText[0]) + token.ValueText.Substring(1);
-                context.RegisterFix(CodeAction.Create($"Rename field to '{newName}'", cancellationToken => RenameHelper.RenameSymbolAsync(document, root, token, newName, cancellationToken)), diagnostic);
+
+                var typeSyntax = this.GetParentTypeDeclaration(token);
+
+                if (typeSyntax != null)
+                {
+                    SemanticModel semanticModel = await document.GetSemanticModelAsync();
+
+                    var typeDeclarationSymbol = semanticModel.GetDeclaredSymbol(typeSyntax) as INamedTypeSymbol;
+                    if (typeDeclarationSymbol == null)
+                    {
+                        return;
+                    }
+
+                    while (!this.IsValidNewMemberName(typeDeclarationSymbol, newName) || newName == typeSyntax.GetTypeName())
+                    {
+                        newName = newName + "Value";
+                    }
+
+                    context.RegisterFix(CodeAction.Create($"Rename field to '{newName}'", cancellationToken => RenameHelper.RenameSymbolAsync(document, root, token, newName, cancellationToken)), diagnostic);
+                }
+            }
+        }
+
+        private bool IsValidNewMemberName(INamedTypeSymbol typeSymbol, string name)
+        {
+            foreach (var member in typeSymbol.GetMembers())
+            {
+                if (member.Kind == SymbolKind.Method)
+                {
+                    // Field names can't colide with method names
```

Comment:
:exclamation: A class is not allowed to have a field and a method with the same name.